sdk/ai/ai-projects/src/agents/files.ts

```diff
@@ -31,6 +31,23 @@ export async function uploadFile(
   return result.body; 
 }
 
+/** Uploads a file for use by other operations. */
+export async function uploadFileAndPoll(
+context: Client, options: UploadFileParameters, sleepInterval: number,
+): Promise<OpenAIFileOutput> {
+  const result = await context.path("/files").post(options);
+  if (!expectedStatuses.includes(result.status)) {
+      throw createRestError(result);
+  }
+  while (result.body.status === "uploaded" || result.body.status === "pending" || result.body.status === "running") {
+    await new Promise(resolve => setTimeout(resolve, sleepInterval));
+    const updatedResult = await context.path(`/files/${result.body.id}`, result.body.id).get(options);
```

Comment:
And then here you'd do `file = await getFile(context, file.id)`

Reply:
Thanks for the suggestion! Updated the operation to remove the redundant code.